modules/EvManager/main/car_simulatorImpl.cpp

```diff
@@ -0,0 +1,586 @@
+    auto valueCopy = value;
+    simData->commandQueue = SimData::parseSimCommands(valueCopy, *commandRegistry);
+
+    if (!simData->commandQueue.empty()) {
+        executionActive = true;
+    }
+}
+
+void car_simulatorImpl::handleSimulationLoop() {
+    while (enabled) {
+        if (simData && executionActive) {
+            if (loopIntervalMs.has_value()) {
+                runSimulationLoop();
+                std::this_thread::sleep_for(std::chrono::milliseconds(loopIntervalMs.value()));
+            } else {
+                break;
+            }
+        }
+    }
+    EVLOG_debug << "Finished simulation.";
+
+    resetSimDataToDefaults();
+
+    // If we have auto_exec_infinite configured, restart simulation when it is done
+    if (mod->config.auto_exec && mod->config.auto_exec_infinite) {
+        auto valueCopy = mod->config.auto_exec_commands;
+        handle_executeChargingSession(valueCopy);
+    }
+}
+
+void car_simulatorImpl::registerAllCommands() {
+    commandRegistry = std::make_unique<CommandRegistry>(this);
+
+    const auto sleepFunction = [](car_simulatorImpl* const simulator, const std::vector<std::string>& arguments) {
+        if (!simulator->sleepTicksLeft.has_value()) {
+            const auto sleepTime = std::stoll(arguments[0]);
+            const auto sleepTimeMs = sleepTime * 1000;
+            simulator->sleepTicksLeft = static_cast<long long>(static_cast<double>(sleepTimeMs) /
+                                                               static_cast<double>(simulator->loopIntervalMs.value())) +
+                                        1;
+        }
+        simulator->sleepTicksLeft = simulator->sleepTicksLeft.value() - 1;
+        return (!(simulator->sleepTicksLeft > 0));
+    };
+    commandRegistry->registerCommand("sleep", 1, sleepFunction);
+
+    const auto iec_wait_pwr_readyFunction = [](car_simulatorImpl* const simulator,
+                                               const std::vector<std::string>& arguments) {
+        return (simulator->simData->pwm_duty_cycle > 7.0f && simulator->simData->pwm_duty_cycle < 97.0f);
+    };
+    commandRegistry->registerCommand("iec_wait_pwr_ready", 0, iec_wait_pwr_readyFunction);
+
+    const auto iso_wait_pwm_is_runningFunction = [](car_simulatorImpl* const simulator,
+                                                    const std::vector<std::string>& arguments) {
+        simulator->simData->state = SimState::PLUGGED_IN;
+        return (simulator->simData->pwm_duty_cycle > 4.0f && simulator->simData->pwm_duty_cycle < 97.0f);
+    };
+    commandRegistry->registerCommand("iso_wait_pwm_is_running", 0, iso_wait_pwm_is_runningFunction);
+
+    const auto draw_power_regulatedFunction = [](car_simulatorImpl* const simulator,
+                                                 const std::vector<std::string>& arguments) {
+        simulator->mod->r_ev_board_support->call_set_ac_max_current(std::stod(arguments[0]));
+        if (arguments[1] == "3") {
```

Comment:
`"3"` define as const

Reply:
Sorry, I'm not sure what you mean. Could you give me an example?

Reply:
E.g. `constexpr auto THREE_PHASES = "3"` -> `if (arguments[1] == THREE_PHASES) {` I find that easier to understand.

Reply:
Alright, I see! I changed that